src/lavinmq/clustering/controller.cr

```diff
@@ -24,25 +24,13 @@ class LavinMQ::Clustering::Controller
   # Each node in a cluster has an unique id, use from config if set,
   # otherwise use stored id on disk, if there's none there either generate one from etcd
   private def clustering_id : Int32
-    id = @config.clustering_id
-    if id < 0
-      id_file_path = File.join(@config.data_dir, ".clustering_id")
-      begin
-        id = File.read(id_file_path).to_i
-      rescue File::NotFoundError
-        loop do
-          max_id = @etcd.get("#{@config.clustering_etcd_prefix}/max_id")
-          id = (max_id || "1000").to_i + 1
-          break if @etcd.put("#{@config.clustering_etcd_prefix}/max_id", id.to_s) == max_id
-        end
-        Dir.mkdir_p @config.data_dir
-        File.write(id_file_path, id.to_s)
-      end
-      @config.clustering_id = id
-    end
-    max_id = @etcd.get("#{@config.clustering_etcd_prefix}/max_id") || "-1"
-    if max_id.to_i < id
-      @etcd.put("#{@config.clustering_etcd_prefix}/max_id", id.to_s)
+    id_file_path = File.join(@config.data_dir, ".clustering_id")
+    begin
+      id = File.read(id_file_path).to_i
+    rescue File::NotFoundError
+      id = rand(Int32::MAX)
```

Comment:
I guess there is some risk of collision here,Why not UUID?

Reply:
I would say it's a tolerable risk, only have to be unique within a cluster, of say 3 or 5 nodes, that's a 0,0000002% risk of collision.